### PR TITLE
Adding Time-Series Info to GCE usage logs

### PR DESCRIPTION
**Pull request by kardiff18**
Worked with Oliver (original author) to add an additional view to allow users to view GCE capacity logs at point-in-time rather than just granular audit-log data.

Includes information on how to generate this interval view as well as how to create a time-series graph in data studio to view the data.

### CODE REVIEW

**Comment by kardiff18** [comment]
@jaketf moved/merged all the files I believe, feel free to take another look. thanks!

File: tools/gce-usage-log/README.md [hunk 1]
```diff
@@ -182,10 +197,116 @@ bq mk \
...
+
+Now, create the view.
+````
+bq query \
+--project $PROJECT_ID \
+--use_legacy_sql=false \
+--destination_table=gce_usage_log.$DESTINATION_TABLE \
+--display_name="Interval usage of GCE Usage Logs" \
+--replace=true \
+--schedule='every 24 hours' $INTERVAL_VIEW
```
Comment (jaketf) [inline tools/gce-usage-log/README.md]: nit: use bash string literal.
```suggestion
--schedule='every 24 hours' "$INTERVAL_VIEW"
```

File: tools/gce-usage-log/README.md [hunk 2]
```diff
@@ -182,10 +197,116 @@ bq mk \
...
+### 3.2.1 How to Use the Interval View
+
 Now that your dataset is ready, how do you query it?
 
+The interval view allows to see aggregated point-in-time statistics. To find resource usage for a specific time-frame, you can query for the aggregate usage information for the month of 
+September by using the sample query below as an example. You can adjust the timeframe by altering the WHERE clause
+or choose to only select certain fields depending on what you are trying to predict.
+
+
+```bash
```
Comment (jaketf) [inline tools/gce-usage-log/README.md]:
```suggestion
```sql
```

File: tools/gce-usage-log/README.md [hunk 3]
```diff
@@ -182,10 +197,116 @@ bq mk \
...
+
+
+Next, create a reference to the interval view:
+
+```
+export INTERVAL_VIEW=$(cat gce_interval_view.sql | sed -e "s/_PROJECT_/$PROJECT_ID/g" -e "s/_TIME_INTERVAL_UNIT_/$TIME_INTERVAL_UNIT/g" -e "s/_TIME_INTERVAL_AMOUNT_/$TIME_INTERVAL_AMOUNT/g")
+```
+
+Now, create the view.
+````
```
Comment (jaketf) [inline tools/gce-usage-log/README.md]: nit 3 backticks bash highlighting
```suggestion
```bash
```

File: tools/gce-usage-log/README.md [hunk 4]
```diff
@@ -182,10 +197,116 @@ bq mk \
...
+bq query \
+--project $PROJECT_ID \
+--use_legacy_sql=false \
+--destination_table=gce_usage_log.$DESTINATION_TABLE \
+--display_name="Interval usage of GCE Usage Logs" \
+--replace=true \
+--schedule='every 24 hours' $INTERVAL_VIEW
+````
```
Comment (jaketf) [inline tools/gce-usage-log/README.md]: nit 3 backticks
```suggestion
```
```

File: tools/gce-usage-log/gce_interval_view.sql [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  tags,
+  labels
+FROM
+  timestamp_interval_table,
+  UNNEST(custom_interval_array) AS custom_interval
+JOIN
+  `_PROJECT_.gce_usage_log._gce_usage_log` usage_view
+ON
+  usage_view.instance_id = timestamp_interval_table.instance_id
+ORDER BY custom_interval asc
```
Comment (jaketf) [inline tools/gce-usage-log/gce_interval_view.sql]: nit
```suggestion
ORDER BY custom_interval ASC
```

File: tools/gce-usage-log/README.md [hunk 6]
```diff
@@ -182,10 +197,116 @@ bq mk \
...
+![interval_query](images/interval-query-results.png)
+
+### 3.2.2 How to Create a Time-Series Graph on the Interval View
+
+If you want to see the same data in a time-series graph rather than a data table, you can do this 
+in Data Studio. This allows you to create time-series graphs to monitor changes and spikes of inventory 
+over time. This can be done on whichever metrics that your team would like to use for capacity planning, 
+such as looking at the total count of instances over time, cores, memory, etc.
+
+1. To create a time-series graph, open up [Data Studio](https://www.datastudio.google.com).
```
Comment (jaketf) [inline tools/gce-usage-log/README.md]: could this be simplified by having a link to an existing public dashboard where user only has to change the data source ?
Similar to [this asset](https://github.com/GoogleCloudPlatform/professional-services/tree/master/examples/bigquery-audit-log#3-copying-the-data-source-in-data-studio)

Comment (kardiff18) [reply tools/gce-usage-log/README.md]: Looking into this, thanks!

**Pull request merged**